src/aero/core.clj

```diff
@@ -34,7 +34,8 @@
          config
          (with-open [pr (java.io.PushbackReader. (io/reader r))]
            (edn/read
-            {:readers (readers {:profile (or profile :default)
+            {:eof {}
```

Comment:
What about returning nil instead? {} is truthy, and it would get through a when clause.  A nil could be merged safely with other maps as you are doing.

Any reason not to use nil?